### ISSUE TITLE
Internal syntect renderer has problems on windows with csharp files that have lines starting with # and background painting

### ISSUE THREAD

**Issue by mitchcapper** [rhysd/hgrep#19]
very specific problem, but not that hard to run into.  I noticed when (h)greping for a specific macro the matching line was just a gray line.

With some more troubleshooting it seems this happens (gray line/whatever highlight color I would guess) any time the background is painted on a line starting with a `#` even when it is not the highlighted result.

Changing the file extension to .c instead of .cs fixes it,  running the same command in linux works fine.  Just about any combo other than a .cs file, or without the line background painted, or without the line starting with a # or not on windows seems ok:)

Sadly I don't see an immediate way to actually turn of background painting for the matched line (other than just using the other printer which does solve this:)).

Windows samples:
![image](https://github.com/rhysd/hgrep/assets/1643324/59060a3a-ddb0-4a4f-b9db-ed68ca2ed41f)

linux samples:
![image](https://github.com/rhysd/hgrep/assets/1643324/c2d9d635-0694-475d-9254-7d06195ab4dc)

**Comment by rhysd** [comment]
I tried this but it didn't reproduce in my local machine.

![image](https://github.com/rhysd/hgrep/assets/823277/83138040-234b-419b-870e-7eca12e7b568)

Some hidden pre-condition would be expected such as:

- Terminal version (I'm using Windows Terminal 1.19.10573.0)
- hgrep version (I'm using the latest version v0.3.5)

This might be an issue of syntax highlighting and some syntax sets were updated on v0.3.5. The update might solve this issue.
